backend.go

```diff
@@ -145,15 +146,28 @@ func (b *jwtAuthBackend) jwtValidator(config *jwtConfig) (*jwt.Validator, error)
 
 	var err error
 	var keySet jwt.KeySet
+	var keySets []jwt.KeySet
 
 	// Configure the key set for the validator
 	switch config.authType() {
 	case JWKS:
 		keySet, err = jwt.NewJSONWebKeySet(b.providerCtx, config.JWKSURL, config.JWKSCAPEM)
+		keySets = []jwt.KeySet{keySet}
+	case MultiJWKS:
+		for k, v := range config.JWKSPairs {
+			// TODO (@johnlanda): read this in as map[string]string in config so we don't need type conversion.
```

Comment:
Is there a pattern for nested config in the path config? Ideally I would like these to be more like a list of structs than a map of key-value pairs.

Reply:
Unfortunately there isn't a great pattern for complex/nested data types in the Vault API. You can use `framework.TypeMap` as you have, but it has the limitation that it's not usable by the Vault CLI without converting the entire request body to JSON. Happy to show you an example of this limitation.

I'm wondering if using `framework.TypeKVPairs` would be better. That would make the data type `map[string]string` where the key is the JWKS URL and the value the CA PEM. It also allows for using the Vault CLI more naturally.

Reply:
`TypeKVPairs` sounds like an improvement over the current setup at least. I don't think the config files will read super intuitively unfortunately since the fields within each `jwks_pair` will be effectively unlabeled.

Reply:
Ya, I see what you're saying. Do you have a rough example of what that config file would look like? It might help me think of something better 🤔

Reply:
If there is anyway to make it like a repeated block in yaml where it would be something like 
```
jwks_pairs:
  - jwks_url: www.foobar.com/wellknown
    jwks_ca_pem: adfjkafklasdfs232flfdskja
  - jwks_url: www.barbaz.com/wellknown
    jwks_ca_pem: aweeqrrnrqewrkljerwqwr
```

Reply:
Looking through the nomad docs for setting up vault doesn't show setting the ca cert. Is that something that vault does automatically? If so we could just make this a list of urls

Reply:
I'm trying something out with the repeated block idea. I'll follow up here shortly!

> Looking through the nomad docs for setting up vault doesn't show setting the ca cert. Is that something that vault does automatically? If so we could just make this a list of urls

Able to share a link to which doc you're looking at? It's not required if your IdP (Nomad in this case) is not TLS protected or the system trust store where Vault is running has already trusted the CA of the IdP.

Reply:
> Looking through the nomad docs for setting up vault doesn't show setting the ca cert. Is that something that vault does automatically?

Even if Nomad doesn't now, or maybe doesn't in its "easy getting starts commands / docs", setting the CA cert might be something we want to support (even if some people leave it as an empty string). I defer to you both, though.

Reply:
Updated so that the config for `jwks_pairs` is effectively a slice of structs and the fields are named for easier reading of config.

Reply:
New config looks like this
```vault write auth/jwt/config -<<EOF 
{
    "jwks_pairs": [
        { "jwks_url": "foobar.com", "jwks_ca_pem": "asdfadsfasdf" },
        { "jwks_url": "barbaz.com", "jwks_ca_pem": "adfsasdfasdf" }
    ]
}
EOF
```

Reply:
Thanks for the example, @johnlanda!